[WIP] Add E2E test for ScyllaDBManagerTask and ScyllaDBCluster integration

czeslavo opened the pull request:
<!-- Please take a look at our [Contributing](https://github.com/scylladb/scylla-operator/blob/master/CONTRIBUTING.md)
documentation before submitting a Pull Request!
Thank you for contributing to the Scylla Operator! -->

**Description of your changes:**

**Which issue is resolved by this Pull Request:**
Resolves #

czeslavo commented:
/test images e2e-gke-multi-datacenter-parallel

czeslavo commented:
/test images e2e-gke-multi-datacenter-parallel

czeslavo commented:
/test images e2e-gke-multi-datacenter-parallel

czeslavo commented:
/test images e2e-gke-multi-datacenter-parallel

rzetelskik commented on test/e2e/set/scylladbmanagertask/scylladbmanagertask_globalmanager_object_storage.go:
```diff
@@ -33,7 +36,454 @@ import (
+				},
+			},
+		}
+		smClusterRegistration, err = nsClient.ScyllaClient().ScyllaV1alpha1().ScyllaDBManagerClusterRegistrations(ns.Name).Create(ctx, smClusterRegistration, metav1.CreateOptions{})
```
hint: smcr is an internal resource, it shouldn't be created directly but by labelling the ScyllaDBCluster. This part should be pretty much identical to the existing test for ScyllaDBDatacenter

czeslavo commented:
@rzetelskik I've addressed your comments and left `// TODO` comments on what is left to be done. Also rebased onto https://github.com/scylladb/scylla-operator/pull/2780.

I got stuck on the `can't create keyspace: gocql: no response received from cassandra within timeout period (potentially executed: false)` error while inserting data (a TODO is left in the code). It was happening for me in a GKE cluster, with an E2E pod running inside the cluster. I didn't manage to debug this.

rzetelskik commented:
Closing in favour of https://github.com/scylladb/scylla-operator/pull/2780.
